tests/test_post_live_metrics.py

```diff
@@ -26,11 +29,107 @@ def test_get_url(monkeypatch, tmp_path_factory):
-def test_studio_token_envvar(monkeypatch, var):
-    monkeypatch.setenv(var, "FOO_TOKEN")
-    monkeypatch.setenv(STUDIO_REPO_URL, "FOO_REPO_URL")
-    assert get_studio_token_and_repo_url() == ("FOO_TOKEN", "FOO_REPO_URL")
+@pytest.mark.parametrize(
+    "token,repo_url",
+    [(DVC_STUDIO_TOKEN, DVC_STUDIO_REPO_URL), (STUDIO_TOKEN, STUDIO_REPO_URL)],
+)
+def test_studio_config_envvar(monkeypatch, token, repo_url):
+    monkeypatch.setenv(token, "FOO_TOKEN")
+    monkeypatch.setenv(repo_url, "FOO_REPO_URL")
+    assert get_studio_config() == {
+        "studio_token": "FOO_TOKEN",
+        "studio_repo_url": "FOO_REPO_URL",
+        "studio_url": STUDIO_URL,
+    }
+
+
+def test_studio_config_dvc_config(monkeypatch):
+    monkeypatch.setenv(DVC_STUDIO_REPO_URL, "FOO_REPO_URL")
+    dvc_config = {
+        "token": "FOO_TOKEN",
+        "url": "FOO_URL",
+    }
+    expected = {
+        "studio_token": "FOO_TOKEN",
+        "studio_repo_url": "FOO_REPO_URL",
+        "studio_url": "FOO_URL",
+    }
+    assert get_studio_config(dvc_config=dvc_config) == expected
+
+
+def test_studio_config_kwarg(monkeypatch):
+    expected = {
+        "studio_token": "FOO_TOKEN",
+        "studio_repo_url": "FOO_REPO_URL",
+        "studio_url": "FOO_URL",
+    }
+    assert (
+        get_studio_config(
+            studio_token="FOO_TOKEN",
+            studio_repo_url="FOO_REPO_URL",
+            studio_url="FOO_URL",
+        )
+        == expected
+    )
+
+
+def test_studio_config_envvar_override(monkeypatch):
+    monkeypatch.setenv(DVC_STUDIO_TOKEN, "FOO_TOKEN")
+    monkeypatch.setenv(DVC_STUDIO_URL, "FOO_URL")
+    monkeypatch.setenv(DVC_STUDIO_REPO_URL, "FOO_REPO_URL")
+    dvc_config = {
+        "token": "BAR_TOKEN",
+        "url": "BAR_URL",
+    }
+    expected = {
+        "studio_token": "FOO_TOKEN",
+        "studio_repo_url": "FOO_REPO_URL",
+        "studio_url": "FOO_URL",
+    }
+    assert get_studio_config(dvc_config=dvc_config) == expected
+
+
+def test_studio_config_kwarg_override(monkeypatch):
+    monkeypatch.setenv(DVC_STUDIO_TOKEN, "FOO_TOKEN")
+    monkeypatch.setenv(DVC_STUDIO_REPO_URL, "FOO_REPO_URL")
+    monkeypatch.setenv(DVC_STUDIO_URL, "FOO_URL")
+    expected = {
+        "studio_token": "BAR_TOKEN",
+        "studio_repo_url": "BAR_REPO_URL",
+        "studio_url": "BAR_URL",
+    }
+    assert (
+        get_studio_config(
+            studio_token="BAR_TOKEN",
+            studio_repo_url="BAR_REPO_URL",
+            studio_url="BAR_URL",
+        )
+        == expected
+    )
+
+
+def test_studio_config_offline(monkeypatch):
+    monkeypatch.setenv(DVC_STUDIO_TOKEN, "FOO_TOKEN")
+    monkeypatch.setenv(DVC_STUDIO_REPO_URL, "FOO_REPO_URL")
+
+    assert get_studio_config() != {}
+
+    assert get_studio_config(offline=True) == {}
+
+    monkeypatch.setenv(DVC_STUDIO_OFFLINE, "true")
+    assert get_studio_config() == {}
+
+    monkeypatch.setenv(DVC_STUDIO_OFFLINE, "1")
```

Comment:
minor: the other supported variations are not tested

Reply:
Done